lib/src/install.rs

```diff
@@ -1058,6 +1093,63 @@ async fn prepare_install(
     Ok(state)
 }
 
+// Backing implementation of --copy-etc; just your basic
+// recursive copy algorithm.  Parent directories are
+// created as necessary
+fn copy_unmanaged_etc(
+    sepolicy: &ostree::SePolicy,
+    src: &Dir,
+    dest: &Dir,
+    path: &mut Utf8PathBuf,
+) -> Result<u64> {
+    let mut r = 0u64;
+    for ent in src.read_dir(&path)? {
+        let ent = ent?;
+        let name = ent.file_name();
+        let name = if let Some(name) = name.to_str() {
+            name
+        } else {
+            anyhow::bail!("Non-UTF8 name: {name:?}");
+        };
+        let meta = ent.metadata()?;
+        path.push(Utf8Path::new(name));
+        r += 1;
+        if meta.is_dir() {
+            if let Some(parent) = path.parent() {
+                dest.create_dir_all(parent)
+                    .with_context(|| format!("Creating {parent}"))?;
+            }
+            if !dest.try_exists(&path)? {
+                let mut db = DirBuilder::new();
+                db.mode(meta.mode());
+                let label = Labeler::new(sepolicy, path, meta.mode())?;
+                dest.create_dir_with(&path, &db)
+                    .with_context(|| format!("Creating {path:?}"))?;
+                drop(label);
+            }
+            r += copy_unmanaged_etc(sepolicy, src, dest, path)?;
+        } else {
+            dest.remove_file_optional(&path)?;
```

Comment:
Or, like Ignition should we default to failing when there is an extant file there?  That gets annoying though because suddenly one needs an extra override flag for when it's truly necessary.